stl/inc/format

```diff
@@ -2557,8 +2557,221 @@ template <_Format_supported_charT _CharT, class _Traits>
 struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
     : _Formatter_base<basic_string_view<_CharT, _Traits>, _CharT, _Basic_format_arg_type::_String_type> {};
 
-using format_context  = basic_format_context<back_insert_iterator<string>, string::value_type>;
-using wformat_context = basic_format_context<back_insert_iterator<wstring>, wstring::value_type>;
+template <class _T>
+class _Fmt_buffer {
+private:
+    _T* _Ptr;
+    size_t _Size;
+    size_t _Capacity;
+
+protected:
+    _Fmt_buffer(size_t _Sz) noexcept : _Size(_Sz), _Capacity(_Sz) {}
+    _Fmt_buffer(_T* _P = nullptr, size_t _Sz = 0, size_t _Cap = 0) noexcept : _Ptr(_P), _Size(_Sz), _Capacity(_Cap) {}
+    ~_Fmt_buffer() = default;
+
+    void set(_T* _Buf_data, size_t _Buf_capacity) noexcept {
+        _Ptr  = _Buf_data;
+        _Size = _Buf_capacity;
+    }
+
+    virtual void grow(size_t _Capacity) = 0;
+
+public:
+    using value_type      = _T;
+    using const_reference = const _T&;
+
+    _Fmt_buffer(const _Fmt_buffer&) = delete;
+    void operator=(const _Fmt_buffer&) = delete;
+
+    _T* begin() noexcept {
+        return _Ptr;
+    }
+    _T* end() noexcept {
+        return _Ptr + _Size;
+    }
+
+    const _T* begin() const noexcept {
+        return _Ptr;
+    }
+    const _T* end() const noexcept {
+        return _Ptr + _Size;
+    }
+
+    size_t size() const noexcept {
+        return _Size;
+    }
+
+    size_t capacity() const noexcept {
+        return _Capacity;
+    }
+
+    _T* data() noexcept {
+        return _Ptr;
+    }
+
+    const _T* data() const noexcept {
+        return _Ptr;
+    }
+
+    void clear() {
+        _Size = 0;
+    }
+
+    void try_resize(size_t _Count) {
+        try_reserve(_Count);
+        _Size = _Count <= _Capacity ? _Count : _Capacity;
+    }
+
+    void try_reserve(size_t _New_capacity) {
+        if (_New_capacity > _Capacity)
+            grow(_New_capacity);
+    }
+
+    void push_back(const _T& _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+
+    template <class _U>
+    void append(const _U* _Begin, const _U* _End);
+
+    template <class _I>
+    _T& operator[](_I _Index) {
+        return _Ptr[_Index];

```

Comment:
Usually we have some `_ITERATOR_DEBUG_LEVEL` checks around such accesses. Not sure whether this is really needed here though